### PR TITLE
add array key to prismic block

### PR DESCRIPTION
**Pull request by IvoElgentos**

### CODE REVIEW

**Review by WouterSteen — CHANGES_REQUESTED** [review]
Is the change in block/Slice.php really needed? b/c seems like we dont do anything with it in that function?

File: Block/Slices.php [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 class Slices extends Group
 {
-    public function fetchItem(\stdClass $slice): string
+    public function fetchItem(\stdClass $slice, $key): string
```
Comment (JeroenBoersma) [inline Block/Slices.php]: :*suggestion* (non-blocking): you could add the `setIterator($key)` here too on the block after `setDocument`

**Review by JeroenBoersma — APPROVED** [review]
Thank you!

Nice submission, added a suggestion but looks good to me

**Comment by JeroenBoersma** [comment]
> Is the change in block/Slice.php really needed? b/c seems like we dont do anything with it in that function?

yes, thought I could imagine that need to set a default for the field for backwards compatibility.

`$key = null;` to make sure that extension will not break.

effective
```php
function fetchItem(..., $key = null)
//...
null !== $key && $block->setIterator($key)
```

**Comment by JeroenBoersma** [comment]
> Is the change in block/Slice.php really needed? b/c seems like we dont do anything with it in that function?

Yes because it overwrites the Grouped and those need to be compatible...

Can you review Wouter? Than we can merge this and tag it as a minor update :beers:

**Review by WouterSteen — APPROVED** [review]

**Pull request merged**